types/ember-data/index.d.ts

```diff
@@ -118,11 +118,11 @@ declare module "ember-data" {
              * Builds a URL for a given type and optional ID.
              */
             buildURL(
-                modelName: string,
-                id: string | any[] | {},
-                snapshot: Snapshot | any[],
-                requestType: string,
-                query: {}
+                modelName?: string,
+                id?: string | any[] | {} | null,
+                snapshot?: Snapshot | any[] | null,
+                requestType?: 'findRecord' | 'findAll' | 'query' | 'queryRecord' | 'findMany' | 'findHasMany' | 'findBelongsTo' | 'createRecord' | 'updateRecord' | 'deleteRecord' | string,
```

Comment:
This reduces to just `string`, so the extra types aren't actually useful here – alas, they don't even give you hints in your editor, unfortunately:

![string](https://user-images.githubusercontent.com/2403023/34423864-70051668-ebdc-11e7-8700-143e48e7b659.png)

Reply:
Oh no! I had added `| string` as an afterthought and didn't check that autocomplete still worked. Will revert.